.github/workflows/renovate.yml

```diff
@@ -3,16 +3,41 @@ name: Renovate
 on:
   # Temporarily enable manual workflow trigger for testing
   workflow_dispatch: {}
+  pull_request:
+    paths:
+      - .github/workflows/renovate.yml
   schedule:
     - cron: "5 2 * * *"
 
+# We are using OpenID Connect to authenticate with Azure with secret.
+# https://docs.github.com/en/actions/security-for-github-actions/security-hardening-your-deployments/configuring-openid-connect-in-azure
+permissions:
+  id-token: write
+  contents: read
+
 jobs:
   renovate:
-    runs-on: ubuntu-latest
+    runs-on: [self-hosted, idp]
+    environment: dev
+
     steps:
       - name: Checkout repository
         uses: actions/checkout@v4
 
+      - name: 'Azuze CLI login'
+        uses: azure/login@v2
+        with:
+          client-id: ${{ vars.AZURE_CLIENT_ID }}
+          tenant-id: ${{ vars.AZURE_TENANT_ID }}
+          subscription-id: ${{ vars.AZURE_SUBSCRIPTION_ID }}
+
+      - name: Get Secrets from Azure Key Vault
+        id: get_secrets
+        run: |
+          # Set env.RENOVATE_TOKEN
+          RENOVATE_TOKEN=$(az keyvault secret show --vault-name ${{ vars.IDP_SM_DEV_KEYVAULT_NAME }} --name "GITHUB-WORKLEAP-PAT" --query value -o tsv)
```

Comment:
TODO: Do not forget to switch back to RENOVATE_TOKEN once the IT ticket is closed and everything is sync to KV